src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -168,13 +168,32 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
             Visit(node.Body);
         }
 
+        public override void VisitRazorMetaCode(RazorMetaCodeSyntax node)
+        {
+            switch (node.Kind)
+            {
+                // These are Braces for things like "@code {}" and "@functions {}".
+                // Hypothetically the braces are the only metacode which should ever reach this,
+                // but I added the switch statement just in case.
+                case SyntaxKind.RazorMetaCode:
+                    AddSemanticRange(node, RazorSemanticTokensLegend.RazorTransition);
+                    break;
+            }
+        }
+
         public override void VisitRazorDirectiveBody(RazorDirectiveBodySyntax node)
         {
             // We can't provide colors for CSharp because if we both provided them then they would overlap, which violates the LSP spec.
             if (node.Keyword.Kind != SyntaxKind.CSharpStatementLiteral)
             {
                 AddSemanticRange(node.Keyword, RazorSemanticTokensLegend.RazorDirective);
             }
+            else
+            {
+                Visit(node.Keyword);
+            }
+
+            Visit(node.CSharpCode);
```

Comment:
lol @ajaybhargavb do we also store the HTML body in this "CSharpCode" member for `@section` directives?

Reply:
Yes if it contains mixed content. I guess the naming of this property is sort of misleading